source/DataAccess/Oracle/OracleDDLHelper.ts

```diff
@@ -28,6 +38,49 @@ export default class OracleDDLHelper implements IDDLHelper {
                 // `SELECT 2 as "priority", TABLE_NAME, DBMS_METADATA.GET_DDL(\'INDEX\', uidx.INDEX_NAME) as ddl ` +
                 // `FROM ALL_INDEXES uidx where TABLE_NAME = '${name}' `;
             })
-            .join('\nunion all\n');
+            .join('\nUNION ALL\n');
+    }
+
+    /**
+     * Determines the order in which database tables should be created
+     * based on their dependency graph.
+     *
+     * @private
+     * @param {string[]} validTableNames All valid tables referenced by the integration type
+     * @returns {Promise<string[]>} A sorted list of objects in order of creation priority
+     * @memberof OracleDDLHelper
+     */
+    private async prioritizeTables(validTableNames: string[]): Promise<string[]> {
+        const connection: oracledb.IConnection = await this._connectionPool.getConnection();
+
+        const tableConstraintQueries = validTableNames.map((tableOrViewName) => {
+            return `SELECT
+                cons.TABLE_NAME as child_table,
+                col.TABLE_NAME parent_table
+            FROM
+                ALL_CONS_COLUMNS col
+                JOIN ALL_CONSTRAINTS cons
+                ON cons.R_OWNER = col.OWNER
+                AND cons.R_CONSTRAINT_NAME = col.CONSTRAINT_NAME
+            WHERE cons.TABLE_NAME = '${tableOrViewName}'`;
+        })
+        .join('\nUNION ALL\n');
+
+        const records = await connection.execute(tableConstraintQueries);
+
+        const tableGraph = new DepGraph<string>();
+        validTableNames.forEach((tableOrView, idx, array) => tableGraph.addNode(tableOrView));
+
+        // Use a simple 2-tuple to represent an association of two database objects
+        let row: [string, string];
+        for (row of records.rows) {
+            // We don't care about self referencing tables so only worry about
+            // inter-table dependencies
+            if (row[0] !== row[1]) {
```

Comment:
Only when a series of tables creates a circular dependency will we fail to report the correct list of prioritized records. The DepGraph class is also explicitly being created without the circular: true flag and so will fail when there's a nasty non-self referencing circular dependency. We'll need to address that once we find one in the wild.

Reply:
AFAIK, the only way to create a circular in RDBMS is to make a nullable foreign key. I'm unfortunate enough to have seen it used in software design, but it's pretty unlikely we would see a schema that does this.